app/services/payment_provider_customers/adyen_service.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module PaymentProviderCustomers
+  class AdyenService < BaseService
+    def initialize(adyen_customer = nil)
+      @adyen_customer = adyen_customer
+
+      super(nil)
+    end
+
+    def create
+      result.adyen_customer = adyen_customer
+      return result if adyen_customer.provider_customer_id?
+
+      adyen_result = generate_checkout_url
+
+      result.checkout_url = adyen_result.response["url"]
+      result
+    end
+
+    def generate_checkout_url
+      res = client.checkout.payment_links_api.payment_links(payment_link_params)
+
+      SendWebhookJob.perform_later(
+        'customer.checkout_url_generated',
+        customer,
+        checkout_url: res.response["url"]
+      )
+
+      return res
+    rescue Adyen::AdyenError => e
+      deliver_error_webhook(e)
+
+      raise
+    end
+
+    def preauthorise(organization, event)
+      shopper_reference = shopper_reference_from_event(event)
+      payment_method_id = event.dig("additionalData", "recurring.recurringDetailReference")
+
+      @adyen_customer = PaymentProviderCustomers::AdyenCustomer.
+        joins(:customer).
```

Comment:
Layout/DotPosition: Place the . on the next line, together with the method name. (https://rubystyle.guide#consistent-multi-line-chains)